dlog/logger.go

```diff
@@ -84,6 +84,19 @@ type OptimizedLogger interface {
 	UnformattedLogf(level LogLevel, format string, args ...interface{})
 }
 
+// LoggerWithMaxLevel can be implemented by loggers that define a maximum
+// level that will be logged, i.e. if a logger defines a max-level of
```

Comment:
nit: "i.e." is used where you might say "in other words"; "e.g." is for where you might say "for example".